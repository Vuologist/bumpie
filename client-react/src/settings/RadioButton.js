import React, { useState } from "react";
import DynamicButton from "../common/DynamicButton";
import styled from "styled-components";

const Main = styled.div`
  display: flex;
  justify-content: space-between;
  @media (max-width: 1036px) {
    display: grid;
    grid-template-columns: 1fr 1fr;
    justify-content: space-between;
  }
`;

const RadButton = styled(DynamicButton)`
  background-color: ${props => (props.color ? "#2EC4B6" : "#8FE8DF")};
  color: white;
  border-radius: 16px;
  padding: 6px 40px 6px 40px;
  @media (max-width: 1036px) {
    margin: 5px;
    padding: 6px 5px 6px 5px;
  }
`;

const RadioButton = () => {
  const [radioSet, setRadioSet] = useState([true, false, false, false]);

  const radioClick = index => {
    setRadioSet(
      radioSet.map((x, i) => {
        if (index != i) {
          x = false;
        } else {
          x = true;
        }
        return x;
      })
    );
  };

  return (
    <Main>
      <RadButton
        text="bi-weekly"
        onClick={() => radioClick(0)}
        color={radioSet[0] ? true : false}
      />
      <RadButton
        text="monthly"
        onClick={() => radioClick(1)}
        color={radioSet[1] ? true : false}
      />
      <RadButton
        text="quarterly"
        onClick={() => radioClick(2)}
        color={radioSet[2] ? true : false}
      />
      <RadButton
        text="yearly"
        onClick={() => radioClick(3)}
        color={radioSet[3] ? true : false}
      />
    </Main>
  );
};

export default RadioButton;
