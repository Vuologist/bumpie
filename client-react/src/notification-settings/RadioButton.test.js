import React from "react";
import RadioButton from "./RadioButton";
import renderer from "react-test-renderer";

it("renders radio buttons correctly", () => {
  const tree = renderer.create(<RadioButton />).toJSON();
  expect(tree).toMatchSnapshot();
});
