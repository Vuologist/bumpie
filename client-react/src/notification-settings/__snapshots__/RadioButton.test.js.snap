// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders radio buttons correctly 1`] = `
.c1 {
  display: inline-block;
  font-weight: 400;
  color: #212529;
  text-align: center;
  vertical-align: middle;
  cursor: pointer;
  -webkit-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  background-color: transparent;
  border: 1px solid transparent;
  padding: 0.375rem 0.75rem;
  font-size: 1rem;
  border-radius: 0.25rem;
  -webkit-transition: color 0.15s ease-in-out,background-color 0.15s ease-in-out, border-color 0.15s ease-in-out,box-shadow 0.15s ease-in-out;
  transition: color 0.15s ease-in-out,background-color 0.15s ease-in-out, border-color 0.15s ease-in-out,box-shadow 0.15s ease-in-out;
}

.c1:focus {
  outline: 0;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c2 {
  background-color: #2EC4B6;
  color: white;
  border-radius: 16px;
  padding: 6px 40px 6px 40px;
}

.c3 {
  background-color: #8FE8DF;
  color: white;
  border-radius: 16px;
  padding: 6px 40px 6px 40px;
}

@media (max-width:1036px) {
  .c0 {
    display: grid;
    grid-template-columns: 1fr 1fr;
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    -ms-flex-pack: justify;
    justify-content: space-between;
  }
}

@media (max-width:1036px) {
  .c2 {
    margin: 5px;
    padding: 6px 5px 6px 5px;
  }
}

@media (max-width:1036px) {
  .c3 {
    margin: 5px;
    padding: 6px 5px 6px 5px;
  }
}

<div
  className="c0"
>
  <button
    className="c1 c2"
    onClick={[Function]}
  >
    bi-weekly
  </button>
  <button
    className="c1 c3"
    onClick={[Function]}
  >
    monthly
  </button>
  <button
    className="c1 c3"
    onClick={[Function]}
  >
    quarterly
  </button>
  <button
    className="c1 c3"
    onClick={[Function]}
  >
    yearly
  </button>
</div>
`;
